modules/services/window-managers/i3-sway/sway.nix

```diff
@@ -469,13 +491,16 @@ in {
         '';
       };
 
-      systemd.user.targets.sway-session = mkIf cfg.systemdIntegration {
+      systemd.user.targets.sway-session = mkIf cfg.systemd.enable {
         Unit = {
           Description = "sway compositor session";
           Documentation = [ "man:systemd.special(7)" ];
           BindsTo = [ "graphical-session.target" ];
-          Wants = [ "graphical-session-pre.target" ];
+          Wants = [ "graphical-session-pre.target" ]
```

Comment:
Do we need both the `Wants` and `Before` field?

Reply:
Yes. Putting it into `Wants` is required for the target to start at all, and putting it into `Before` ensures that it only gets started *after* `sway-session.target` is considered active.

This should avoid any race conditions that might arise.

Reply:
@alexarice Who could be this someone? Can we ask someone to have a look? On my try it worked fine but I'm also not an expert in systemd-units neither have merge permissions :)